packages/backend/src/peripherals/database/transactions/UserTransactionRepository.ts

```diff
@@ -123,9 +125,31 @@ export class UserTransactionRepository extends BaseRepository {
       .where('stark_key_a', starkKey.toString())
       .orWhere('stark_key_b', starkKey.toString())
       .orderBy('timestamp', 'desc')
+
+    if (pagination) {
+      query = query.limit(pagination.limit).offset(pagination.offset)
+    }
+
     return toRecords<T>(await query)
   }
 
+  async getCountByStarkKey(
```

Comment:
I am not sure whether this function belongs here. Shouldn't repositories deal only with inserting, editing, deleting and retrieving data?

Reply:
Calculating count is also a query to the database, in this sense it's a "read". I think it's the right place.